### DISCUSSION TITLE
Proper way to get response headers from API

### DISCUSSION THREAD

**Discussion by viiiprock** [urql-graphql/urql#1240] [Q&A] [answered]
I'm using next-urql & exchange-auth packages to do the authentication, my api server automatically refreshed token and responses via headers. Is there any way to get response headers instead of I must implement a custom fetchExchange? (which seems not a good idea solution) 
Thanks for any answer.

**Comment by JoviDeCroock** [comment, marked as answer]
Generally I'd advise to use cookies for this purpose, you could also choose to return the new token as part of the extensions field. You should also consider the point that implicit token refreshes often lead to scenario's where in multiple concurrent calls multiple tokens are refreshed at the same time which can end up leaving the user with an invalidated token.

**Reply by viiiprock** [reply]
It's a bit complicate on my case at the moment, change needs some backend business and time work. In fact, our backend side only generates new token when old token invalidated and auto-removed when expired, not very frequently.  
I temporary use custom fetch exchange, and discuss with team about your idea for the next implement. Thank for your advice!
